modules/tinymce/src/core/test/ts/browser/lookup/UserLookupTest.ts

```diff
@@ -0,0 +1,325 @@
+import { describe, it } from '@ephox/bedrock-client';
+import { Arr, Fun, Obj } from '@ephox/katamari';
+import { TinyHooks } from '@ephox/wrap-mcagar';
+import { expect } from 'chai';
+
+import type Editor from 'tinymce/core/api/Editor';
+import type { User } from 'tinymce/core/lookup/UserLookup';
+
+const createMockUser = (id: string): User => ({
+  id,
+  name: 'Test User',
+  avatar: 'test-avatar.png',
+  description: 'Test Description'
+});
+
+describe('browser.tinymce.core.UserLookupTest', () => {
+  const hook = TinyHooks.bddSetupLight<Editor>({
+    base_url: '/project/tinymce/js/tinymce',
+    user_id: 'test-user-1',
+    fetch_users: (userIds: string[]): Promise<User[]> => {
+      return new Promise((resolve) => {
+        setTimeout(() => {
+          const validIds = Arr.filter(userIds, (id): id is string =>
+            typeof id === 'string' && id.length > 0
+          );
+
+          // Only create users for valid test IDs
+          const users = Arr.bind(validIds, (id) =>
+            id.startsWith('test-user-') ? [ createMockUser(id) ] : []
+          );
+
+          resolve(users);
+        }, 0);
+      });
+    },
+  });
+
+  it('Should return the configured user ID', () => {
+    const editor = hook.editor();
+    const currentUserId = editor.userLookup.getCurrentUserId();
+    expect(currentUserId).to.equal('test-user-1', 'Should return the configured user ID');
+  });
+
+  it('Should fetch users and return array of promises', () => {
+    const editor = hook.editor();
+    const userIds = [ 'test-user-1' ];
+
+    const promises = editor.userLookup.fetchUsers(userIds);
+
+    expect(promises).to.be.an('array', 'Should return an array of promises');
+    expect(promises).to.have.lengthOf(1, 'Should return array with one promise');
+    expect(promises[0]).to.be.instanceOf(Promise, 'Should return a Promise');
+  });
+
+  it('Should return multiple promises for multiple userIds', () => {
+    const editor = hook.editor();
+    const userIds = [ 'test-user-1', 'test-user-2' ];
+
+    const promises = editor.userLookup.fetchUsers(userIds);
+
+    expect(promises).to.have.lengthOf(2, 'Should return array with two promises');
+    expect(promises[0]).to.be.instanceOf(Promise, 'First item should be a Promise');
+    expect(promises[1]).to.be.instanceOf(Promise, 'Second item should be a Promise');
+  });
+
+  it('Should throw an error for empty string ids', async () => {
+    const editor = hook.editor();
+    const userIds = [ '' ];
+    const [ userPromise ] = editor.userLookup.fetchUsers(userIds);
+
+    try {
+      await userPromise;
+      expect.fail('Should throw error for empty string ID');
+    } catch (err: unknown) {
+      expect(err).to.be.instanceOf(Error);
+      expect(err).to.have.property('message', 'User  not found');
+    }
```

Comment:
This isn't what I asked for when I said switch to chai. Chai has an assertion to handle this for you.
https://www.chaijs.com/api/bdd/#method_throw

Although maybe it's not that simple for promises, there's an addon library for it 🤔 
https://www.chaijs.com/plugins/chai-as-promised/